### DISCUSSION TITLE
Best way to have multilingual routes/URLs ?

### DISCUSSION THREAD

**Discussion by ntltd** [vercel/next.js#15522] [Help]
Hi all,

I'm currently migrating an e-commerce CRA/Redux to NextJS and trying to find the best way to keep the existing multilingual routing for SEO purposes.

For example, I have a dynamic route to display recipes. In English the route is `/recipe/:id`, in French it's `/recette/:id` and I also have a shortened one `/r/:id`. `id` is a dynamic parameter associated to a recipe's unique identifier. 

What is the best way to reproduce this using NextJS? I want to do this without creating locales sub-paths (and associated redirections).

All help appreciated 🙏 

---

I've tried different solutions but still struggling which one is the best:

**1. Duplicated pages and paths**
```
pages/
    recipe/
        [id].js
    recette/
        [id].js
    r/
        [id].js
```
Which is very painful and seems to be a lot of duplicated code...

**2. Use dynamic routes**
```
pages/
    [recipe]/
        [id].js
```
Which only works (in my understanding) with `getStaticPaths` but will catch every other pages with the same "logic" (`/my-account/` (EN) === `/mon-compte/` (FR))...

**3. Create a custom server**
Which is something I don't really want because I will loose Automatic Static Optimisation...

**Any other solution**
[YOURS]

**Comment by srikanth912** [comment]
How about using symlinks? Still some repetition (in directory structure) but not as bad as the first option.


```
pages/
    recipe/
        [id].js
    recette/
        [id].js  -> ../recipe/[id].js
    r/
        [id].js  -> ../recipe/[id].js
```

**Reply by ntltd** [reply]
Thanks for you answer!
I'm not very confident with symlinking in a NextJS app, how would you have done that?
